docs/docs/rest-api/public/api/v2/types/pod-instance-status.raml

```diff
@@ -0,0 +1,62 @@
+#%RAML 1.0 DataType
+type: object
+additionalProperties: false
+properties:
+  id:
+    type: !include name.raml
+    description: |
+      Unique ID of this pod instance in the cluster.
+      TODO(jdef) Probably represents the Mesos executor ID.
+  status:
+    type: string
+    enum: [ PENDING, STAGING, STABLE, DEGRADED, TERMINAL ]
```

Comment:
Assuming that you mean "mapping task status to [container] status" because
there's already a plan to aggregate the individual container statuses into
a pod-instance status. If that's the case ... what's the plan? Can you
point me to a PR?


On Fri, Aug 26, 2016 at 10:51 AM, Matthias Eichstedt <
notifications@github.com> wrote:

> In docs/docs/rest-api/public/api/v2/types/pod-instance-status.raml
> <https://github.com/mesosphere/marathon/pull/4250#discussion_r76431667>:
>
> > @@ -0,0 +1,62 @@
> > +#%RAML 1.0 DataType
> > +type: object
> > +additionalProperties: false
> > +properties:
> > +  id:
> > +    type: !include name.raml
> > +    description: |
> > +      Unique ID of this pod instance in the cluster.
> > +      TODO(jdef) Probably represents the Mesos executor ID.
> > +  status:
> > +    type: string
> > +    enum: [ PENDING, STAGING, STABLE, DEGRADED, TERMINAL ]
>
> As noted above, the plan for the internal model is to abstract
> (AppDefinition) task instances and (PodDefinition) instances into one
> Instance. A task can have the following states:
>
> Reserved
> Created
> Error
> Failed
> Finished
> Killed
> Killing
> Running
> Staging
> Starting
> Unreachable
> Gone
> Dropped
> Unknown
>
> For the sake of aligning things in v2, what about simply mapping task
> status to instance status? @unterstein <https://github.com/unterstein>
> made plan of how status are aggregated into an aggregate status of a pod
> instance (e.g., the instance is staging as long as any task is staging, and
> when all are running, the instance turns running).
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4250/files/c1e350f76ab698dd26c9bfba42fb6d14995bc7b5#r76431667>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLOOqtG_vvDGBJvQrw-Fsuu1s10aiks5qjv2DgaJpZM4Jsi1k>
> .
>